packages/compartment-mapper/src/node-modules.js

```diff
@@ -169,23 +169,45 @@ const findPackage = async (readDescriptor, canonical, directory, name) => {
   }
 };
 
-const languages = ['mjs', 'cjs', 'json', 'text', 'bytes'];
-const uncontroversialParsers = {
+const defaultLanguages = /** @type {const} */ ([
+  'mjs',
+  'cjs',
+  'json',
+  'text',
+  'bytes',
+]);
+const defaultUncontroversialParsers = /** @type {const} */ ({
   cjs: 'cjs',
   mjs: 'mjs',
   json: 'json',
   text: 'text',
   bytes: 'bytes',
-};
-const commonParsers = { js: 'cjs', ...uncontroversialParsers };
-const moduleParsers = { js: 'mjs', ...uncontroversialParsers };
+});
+const defaultCommonParsers = /** @type {const} */ ({
+  js: 'cjs',
+  ...defaultUncontroversialParsers,
+});
+const defaultModuleParsers = /** @type {const} */ ({
+  js: 'mjs',
+  ...defaultUncontroversialParsers,
+});
 
 /**
  * @param {object} descriptor
  * @param {string} location
+ * @param {{languages?: readonly string[]|string[], uncontroversialParsers?: Record<string, string>, commonParsers?: Record<string, string>, moduleParsers?: Record<string, string>}} [options]
```

Comment:
probably wants a typedef, but I'm not gonna bother unless someone asks

Reply:
```suggestion
 * @param {object} [options]
 * @param {readonly string[]|string[]} [options.languages]
 * @param {Record<string, string>} [options.uncontroversialParsers]
 * @param {Record<string, string>} [options.commonParsers]
 * @param {Record<string, string>} [options.moduleParsers]
```